Update Lua script documentation for `openspace.navigation` and `openspace.pathnavigation`

WeirdRubberDuck opened the pull request:
To try to take advantage of new documentation functionalities 

To avoid writing really confusing documentation I have also changed the input parameters of:
* `openspace.navigation.addLocalRoll`
* `openspace.navigation.addGlobalRoll`
* `openspace.navigation.addTruckMovement`
to use 1 parameter instead of 2 (where one was not used in the end)

and tried to fix a gnarly return value for the `openspace.navigation.joystickaxis` function. 
❗ **However, documented structs are not supported as return values in codegen, so this needs some discussion** ❗

WeirdRubberDuck commented on src/navigation/navigationhandler_lua.inl:
```diff
@@ -254,43 +285,96 @@ namespace {
     );
 }
 
+// @TODO: Should this be a concrete Type in the documentation? Could also be used in bind
```
Should we do this?  Would require rewriting the [documentation pages on the wiki ](https://docs.openspaceproject.com/en/latest/manual/navigation/joystick-customization.html), so maybe not worth the effort?

WeirdRubberDuck commented on src/navigation/navigationhandler_lua.inl:
```diff
@@ -254,43 +285,96 @@ namespace {
+    std::optional<bool> isRemote;
+};
+
+// @TODO: How to create a reference to the above object as the return type?
```
@ylvaselling is this possible in the new documentation?  I feel like I need to discuss how to nicely handle return values that represent a certain type of objects

WeirdRubberDuck commented:
Drafting as I need to discuss some part of the `@TODO` comments. 

@Roxeena please make sure the joystick_helper file still works as expected.  I have not tried running anything, just rewrote it

WeirdRubberDuck commented on src/navigation/navigationhandler_lua.inl:
```diff
@@ -175,26 +202,29 @@ namespace {
     global::navigationHandler->orbitalNavigator().startRetargetAnchor();
 }
 
+// @TODO: consider adding enum types for axis type and joysticktype
```
Discuss

WeirdRubberDuck commented on src/navigation/navigationhandler_lua.inl:
```diff
@@ -430,14 +572,18 @@ joystickAxis(std::string joystickName, int axis)
 
 /**
  * Return the complete list of connected joysticks
```
```suggestion
 * Return the complete list of connected joysticks.
```

WeirdRubberDuck commented on src/navigation/navigationhandler_lua.inl:
```diff
@@ -430,14 +572,18 @@ joystickAxis(std::string joystickName, int axis)
 
 /**
- * Returns the distance in meters to the current focus node
+ * Return the distance to the current focus node
```
```suggestion
 * Return the distance to the current focus node.
```

WeirdRubberDuck commented on src/navigation/navigationhandler_lua.inl:
```diff
@@ -449,7 +595,9 @@ joystickAxis(std::string joystickName, int axis)
 
 /**
- * Returns the distance in meters to the current focus node's bounding sphere
+ * Return the distance to the current focus node's bounding sphere
```
```suggestion
 * Return the distance to the current focus node's bounding sphere.
```

WeirdRubberDuck commented on src/navigation/navigationhandler_lua.inl:
```diff
@@ -463,7 +611,9 @@ joystickAxis(std::string joystickName, int axis)
 
 /**
- * Returns the distance in meters to the current focus node's interaction sphere
+ * Return the distance to the current focus node's interaction sphere
```
```suggestion
 * Return the distance to the current focus node's interaction sphere.
```

WeirdRubberDuck commented on src/navigation/pathnavigator_lua.inl:
```diff
@@ -24,32 +24,45 @@
     openspace::global::navigationHandler->pathNavigator().skipToEnd();
 }
 
+// @TODO: Split this up into two functions and then document (to remove the confusion)
```
Discuss.  Let me know what you think here

alexanderbock replied:
What would the two functions be in this case?

WeirdRubberDuck replied:
Good question. I had some idea when I wrote this... I think it was to create another function that uses the up-direction, to get signatures like:

`openspace.pathnavigation.flyTo(identifier, std::optional<double> duration)`
`openspace.pathnavigation.flyToNodeAndPreserveUp(identifier, std::optional<double> duration)` (naming here is hard...)

Do they have to have different names though?  Or could we create "overloaded functions"?  Either way, writing the documentation for this seems almost impossible to do in an understandable way

WeirdRubberDuck commented on src/navigation/navigationhandler_lua.inl:
```diff
@@ -175,26 +202,29 @@ namespace {
  * \param shouldInvert if the joystick movement should be inverted or not
  * \param joystickType what type of joystick or axis this is. Either
- *                     <code>"JoystickLike"</code> or <code>"TriggerLike"</code>.
+ *                     `\"JoystickLike\"` or `\"TriggerLike\"`
```
Not sure the `\` are needed here

Roxeena commented on src/navigation/navigationhandler_lua.inl:
```diff
@@ -314,14 +404,21 @@ joystickAxis(std::string joystickName, int axis)
+ *               `"Press"`, which means that the script is run when the user presses the
+ *               button. Alternatives are `"Idle"` (if the button is unpressed and has
+ *               been unpressed since last frame), `"Repeat"` (if the button has been
+ *               pressed since longer than last frame), and `"Release"` (f the button was
```
```suggestion
 *               pressed since longer than the last frame), and `"Release"` (if the button was
```

Roxeena commented on src/navigation/navigationhandler_lua.inl:
```diff
@@ -314,14 +404,21 @@ joystickAxis(std::string joystickName, int axis)
+ * \param action the action for when the script should be executed. This defaults to
+ *               `"Press"`, which means that the script is run when the user presses the
+ *               button. Alternatives are `"Idle"` (if the button is unpressed and has
+ *               been unpressed since last frame), `"Repeat"` (if the button has been
```
```suggestion
 *               been unpressed since the last frame), `"Repeat"` (if the button has been
```

Roxeena commented on src/navigation/navigationhandler_lua.inl:
```diff
@@ -374,43 +479,80 @@ joystickAxis(std::string joystickName, int axis)
+/**
+ * Directly add to the global rotation of the camera (around the focus node).
+ *
+ * \param v1 the value to add in the x-direction (a positive value rotates to the
```
Should we rename these parameters? V1 and V2 are not very descriptive.

WeirdRubberDuck replied:
I considered that, but was unsure about the naming (e.g. is it a velocity or value? ) so left it as is for now.  

Got any suggestion for alternative names?

Roxeena commented on src/navigation/pathnavigator_lua.inl:
```diff
@@ -265,11 +285,14 @@ namespace {
- * sphere away from the object. The optional input parameter specifies the duration for
- * the motion, in seconds.
+ * Fly linearly to a specific distance in relation to the focus node, given as a relative
+ * value based on the size of the objects rather than in meters.
```
```suggestion
 * value based on the size of the object rather than in meters.
```

Roxeena commented on src/navigation/pathnavigator_lua.inl:
```diff
@@ -311,7 +334,9 @@ namespace {
- * If no fade duration is specified, use the property from Navigation Handler
+ * \param navigationState A [NavigationState](#core_navigation_state) to jump to
+ * \param fadeDuration An optional duration for the fading. If not included, the
+ *                     property in Navigation Handler will be used
```
Add a reference to the Navigation Handler maybe

WeirdRubberDuck replied:
@ylvaselling, can we link to properties even? :)  How are the URLs for them created?

WeirdRubberDuck replied:
Actually, the Settings part is not in the generated part of the new documentation page. 

@ylvaselling Maybe let's add it as another menu under "Reference"?

Roxeena commented on src/navigation/navigationhandler_lua.inl:
```diff
@@ -25,30 +25,39 @@
+ * json format, such as the output files of saveNavigationState.
+ *
+ * \param filePath the path to the file, including the file name (and extension, if it is
+ *                 anything alse than `.navstate`)
```
```suggestion
 *                 anything other than `.navstate`)
```

Roxeena commented on src/navigation/navigationhandler_lua.inl:
```diff
@@ -25,30 +25,39 @@
- * set to the scene graph root.
+ * Return the current navigation state as a Lua table.
+ *
+ * By default, the reference frame will picked based on whether the orbital navigator is
```
```suggestion
 * By default, the reference frame will be picked based on whether the orbital navigator is
```

Roxeena reviewed (commented):
I have not tested it yet with controllers but the code looks good

WeirdRubberDuck commented on src/navigation/navigationhandler_lua.inl:
```diff
@@ -25,30 +25,39 @@
- * of saveNavigationState. If usetimeStamp is set to true and the provided navigation
- * state has a timestamp, time will be set as well.
+ * Set the camera position by loading a navigation state from file. The file should be in
+ * json format, such as the output files of saveNavigationState.
```
Code snippet, yes 👍    Not sure if we can refer to it

WeirdRubberDuck commented on src/navigation/navigationhandler_lua.inl:
```diff
@@ -175,26 +202,29 @@ namespace {
+ * Bind an axis of a joystick to be used as a certain type.
+ *
+ * There are also some extra settings that control the behavior:
+ * - If `isInverted` is `true`, the axis value is inverted.
```
Should be the same

WeirdRubberDuck commented on src/navigation/navigationhandler_lua.inl:
```diff
@@ -254,43 +285,96 @@ namespace {
- * bound to the axis as string (empty is type is not Property), the min and max values for
- * the property as numbers and whether the property change will be executed remotly as
- * bool.
+ * Return all the information bound to a certain joystick axis.
```
That should be apparent from the return value.  Ideally, I would want to use a reference to the object above (that's one of my points for discussion)

WeirdRubberDuck replied:
changed the `\return` information below to make it clearer that the returned value is "an object"

WeirdRubberDuck commented on src/navigation/navigationhandler_lua.inl:
```diff
@@ -25,30 +25,39 @@
- * of saveNavigationState. If usetimeStamp is set to true and the provided navigation
- * state has a timestamp, time will be set as well.
+ * Set the camera position by loading a navigation state from file. The file should be in
+ * json format, such as the output files of saveNavigationState.
```
```suggestion
 * json format, such as the output files of `saveNavigationState`.
```

WeirdRubberDuck commented on src/navigation/navigationhandler_lua.inl:
```diff
@@ -143,8 +168,10 @@ namespace {
-// If the current anchor is not an interesting node, the first will be selected
+/**
+ * Picks the next node from the interesting nodes out of the profile and selects that. If
+ * the current anchor is not an interesting node, the first will be selected
```
Not sure what you mean with this comment

WeirdRubberDuck commented on src/navigation/navigationhandler_lua.inl:
```diff
@@ -25,30 +25,39 @@
+ * \param frame the identifier of an optional scene graph node to use as reference frame
+ *              for the navigation state
+ *
+ * \return a Lua table representing the current [navigation state](#core_navigation_state)
```
should we use `NavigationState` here or stick with `navigation state`?  We're currently inconsistent

WeirdRubberDuck commented on src/navigation/navigationhandler_lua.inl:
```diff
@@ -72,9 +81,14 @@ namespace {
+/**
+ * Set the camera position from a provided navigation state.
+ *
+ * \param navigationState a table describing the [NavigationState](#core_navigation_state)
```
see previous comment

WeirdRubberDuck commented:
Requires a test from @Roxeena, but then this PR should be ready for merge.

I addressed all your comments and removed any TODOs that I had left in there. Doing any larger changes than what's currently in here does not seem suitable for the upcoming release.

Roxeena commented:
I tested it with the Xbox and SpaceMouse asset now and it works as it should :+1:
